### PR TITLE
Service alias + relocate vault path validation

### PR DESCRIPTION
**Pull request by akanto**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/kerberos/KeytabProvider.java [hunk 1]
```diff
@@ -26,7 +26,12 @@ public ServiceKeytabResponse getServiceKeytabResponse(Stack stack, GatewayConfig
             throws Exception {
         ServiceKeytabRequest request = new ServiceKeytabRequest();
         request.setEnvironmentCrn(stack.getEnvironmentCrn());
-        request.setServerHostName(primaryGatewayConfig.getHostname());
+        String hostname = primaryGatewayConfig.getHostname();
+        request.setServerHostName(hostname);
+        int dotIndex = hostname.indexOf('.');
+        if (dotIndex != -1) {
+            request.setServiceAlias(hostname.substring(0, dotIndex));
+        }
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/kerberos/KeytabProvider.java]: StringUtils#substringBefore would be simpler: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#substringBefore-java.lang.String-java.lang.String-

Comment (akanto) [reply core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/kerberos/KeytabProvider.java]: Makes sense, and I will add some unit test

File: freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java [hunk 2]
```diff
@@ -349,7 +360,7 @@ private String getExistingKeytab(String canonicalPrincipal, FreeIpaClient ipaCli
         }
     }
 
-    private static String constructCanonicalPrincipal(String serviceName, String hostName, String realm) {
+    private static String constructPrincipal(String serviceName, String hostName, String realm) {
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java]: `hostName` is a bit misleading as we pass on alias too

Comment (akanto) [reply freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java]: Unfortunately, we use hostname for fqdn and also we use hostname for hostname in many places. I have renamed the to serverHostNameAlias, but consistent naming everywhere in the code is a massive refactor

**Pull request merged**